projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.ts

```diff
@@ -0,0 +1,151 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges} from '@angular/core';
+import {BaseValueComponent} from '../base-value.component';
+import {CreateLinkValue, ReadLinkValue, ReadResource, UpdateLinkValue, KnoraApiConnection} from '@knora/api';
+import {Subscription} from 'rxjs';
+import {FormBuilder, FormControl, FormGroup, Validators} from '@angular/forms';
+import {KnoraApiConnectionToken} from '../../../core';
+@Component({
+  selector: 'kui-link-value',
+  templateUrl: './link-value.component.html',
+  styleUrls: ['./link-value.component.scss']
+})
+export class LinkValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+  @Input() displayValue?: ReadLinkValue;
+  options: ReadResource[];
+  restrictToResourceClass: string;
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  form: FormGroup;
+  linkedResource: ReadResource;
+  linkedResourceIRI: string;
+
+  valueChangesSubscription: Subscription;
+  // label cannot contain logical operations of lucene index
+  customValidators = [];
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, @Inject(KnoraApiConnectionToken) private knoraApiConnection: KnoraApiConnection) {
+    super();
+  }
+
+  /**
+   * Search for resources whose labels contain the given search term, restricting to to the given properties object constraint.
+   * this is to be used for update and new linked resources
+   *
+   * @param label to be searched
+   */
+    searchByLabel(searchTerm: string): ReadResource[] {
+      this.restrictToResourceClass = this.displayValue.linkedResource.type;
+      // at least 3 characters are required
+      if (searchTerm.length >= 3) {
+
+        this.knoraApiConnection.v2.search.doSearchByLabel(searchTerm, 0, {limitToResourceClass: this.restrictToResourceClass} ).subscribe(
+          (response: ReadResource[]) => {
+            this.options = response;
+          });
+      } else {
+        // clear selection
+        this.options = undefined;
+
+      }
+      return this.options;
+  }
+
+  // show the label of the linked resource
+  getInitValue(): ReadResource | null {
+    if (this.displayValue !== undefined) {
+      return this.displayValue.linkedResource;
+    } else {
+      return null;
+    }
+  }
+  // override the resetFormControl() from the base component to deal with initial link value label
+  resetFormControl(): void {
+    super.resetFormControl();
```

Comment:
I think it is enough to do:

```ts
  // override the resetFormControl() from the base component to deal with initial link value label
  resetFormControl(): void {
    super.resetFormControl();

    if (this.valueFormControl !== undefined) {
      if (this.mode !== 'read') {

        this.valueChangesSubscription = this.valueFormControl.valueChanges.subscribe(data => {
          this.options = this.searchByLabel(data);
        });

      }
    }
  }
```

Please also unsubscribe from this in `ngOnDestroy` (like we do for the comment change subscription).

Perhaps you also need to include this here from `ngOnInit`:

`this.options = [this.displayValue.linkedResource];`

Reply:
well, in resetFromControl, I need to set the label of the linkedResource initial value of the link value as well.

Reply:
Sorry, I don't understand. In `getInitValue` you are returning the linked resource from the link value. Hence this `ReadResource` is set for the `FormControl`'s value (default behavior of `resetFormControl`). Since there is `[displayWith]="displayResource"`, the component knows what to display in the input field (the resource's label).